go/parquet/pqarrow/file_reader.go

```diff
@@ -356,11 +356,9 @@ func (fr *FileReader) ReadRowGroups(ctx context.Context, indices, rowGroups []in
 
 	// pass pairs of reader and column index to the channel for the
 	// goroutines to read the data
-	for idx, r := range readers {
-		defer func(r *ColumnReader) {
-			r.Release()
-		}(r)
-		ch <- readerInfo{r, idx}
+	for idx := range readers {
+		defer readers[idx].Release()
```

Comment:
Cleaned up a little here, shorter and easier to understand, without a shadowing-closure for the defer, and no confusion as of to the value of `r` w.r.t. `range`.